test/Microsoft.OData.ModelBuilder.Tests/Commons/MockType.cs

```diff
@@ -85,8 +85,46 @@ public MockType Property(Type propertyType, string propertyName, params Attribut
             var mockPropertyInfo = new MockPropertyInfo(propertyType, propertyName);
             mockPropertyInfo.SetupGet(p => p.DeclaringType).Returns(this);
             mockPropertyInfo.SetupGet(p => p.ReflectedType).Returns(this);
+
+            // The model builder does checks custom property attributes to support some featuers ([Required], [Contained], etc.)
+            // So we have to mock the GetCustomAttributes methods. There are two overloads that we need worry about.
+            // - object[] GetCustomAttributes(bool inherit)
+            // - object[] GetCustomAttributes(Type attributeType, bool inherit)
+
+            // Let's start with the first overload. It should return all the attributes without
+            // filtering on type.
             mockPropertyInfo.Setup(p => p.GetCustomAttributes(It.IsAny<bool>())).Returns(attributes);
 
+            // The second overload is a bit more complicated. This overload is also called internally
+            // by extension methods like propertyInfo.GetCustomAttribute<T> or GetCustomAttributes<T>()
+            // The overload should only return attributes of the specified type.
+            // We should ensure that we also return an array of the specific attribute type, even if it's empty.
+            // For example, if we call property.GetCustomAttributes<MyAttribute>(), we should return
+            // an instance of MyAttribute[], not an instance of object[] or Attribute[] that only contains MyAttribute values.
+            // This should be the case even if the array is empty. Which means we should mock any possible Attribute type.
+            // This is important because MyAttribute[] can always be causet to Attribute[], but Attribute[] cannot always be cast to MyAttribute[]
```

Comment:
I think you meant `cast` here:
```suggestion
            // This is important because MyAttribute[] can always be cast to Attribute[], but Attribute[] cannot always be cast to MyAttribute[]
```